modules/core/src/main/java/org/apache/ignite/internal/util/CollectionUtils.java

```diff
@@ -204,6 +204,68 @@ public boolean contains(Object o) {
         };
     }
 
+    /**
+     * Union lists.
+     *
+     * @param lists Lists.
+     * @param <T> Type of the elements of lists.
+     * @return Immutable union of lists.
+     */
+    @SafeVarargs
+    public static <T> List<T> union(List<T>... lists) {
+        if (lists == null || lists.length == 0) {
+            return List.of();
+        }
+
+        return new AbstractList<>() {
+            /** {@inheritDoc} */
+            @Override
+            public T get(int index) {
+                for (int i = 0; i < lists.length; i++) {
+                    List<T> list = lists[i];
+
+                    if (index >= list.size()) {
+                        index -= list.size();
+                    } else {
+                        return list.get(index);
+                    }
+                }
+
+                throw new IndexOutOfBoundsException(index);
+            }
+
+            /** {@inheritDoc} */
+            @Override
+            public Iterator<T> iterator() {
+                return concat(lists).iterator();
+            }
+
+            /** {@inheritDoc} */
+            @Override
+            public int size() {
+                int size = 0;
+
+                for (int i = 0; i < lists.length; i++) {
+                    size += lists[i].size();
+                }
+
+                return size;
+            }
+
+            /** {@inheritDoc} */
+            @Override
+            public boolean contains(Object o) {
```

Comment:
Why do you need to implement this method?

Reply:
Less memory will be used.